packages/redeem_solver/src/storage.ts

```diff
@@ -105,6 +108,38 @@ export class D1Storage implements Storage {
 		return results;
 	}
 
+	async processPutRedeemTx(setupId: number, r: RedeemRequestEventRaw): Promise<void> {
```

Comment:
you need to take txid / txDigest and record it as well: https://github.com/gonative-cc/workers/blob/robert%2Fparallel-scheduler/packages/sui-indexer/src/handler.ts#L62